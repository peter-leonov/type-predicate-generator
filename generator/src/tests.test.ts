import { expect, test } from "vitest";
import {
  array,
  combine,
  object,
  union,
  value,
  type Value,
} from "./tests";

test(combine, () => {
  const obj = object<Value>({
    a: union([value(1), value(2)]),
    b: union([value(true), value(false)]),
    c: array(union([value("a"), value("b")])),
    d: object({
      d2: union([value(null), value(undefined)]),
    }),
  });

  expect(combine(obj)).toMatchSnapshot();
});
