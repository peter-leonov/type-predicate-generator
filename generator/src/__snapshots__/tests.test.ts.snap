// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`combine 1`] = `
[
  {
    "a": 1,
    "b": true,
    "c": [
      "a",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 1,
    "b": true,
    "c": [
      "a",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 1,
    "b": true,
    "c": [
      "b",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 1,
    "b": true,
    "c": [
      "b",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 1,
    "b": false,
    "c": [
      "a",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 1,
    "b": false,
    "c": [
      "a",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 1,
    "b": false,
    "c": [
      "b",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 1,
    "b": false,
    "c": [
      "b",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 2,
    "b": true,
    "c": [
      "a",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 2,
    "b": true,
    "c": [
      "a",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 2,
    "b": true,
    "c": [
      "b",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 2,
    "b": true,
    "c": [
      "b",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 2,
    "b": false,
    "c": [
      "a",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 2,
    "b": false,
    "c": [
      "a",
    ],
    "d": {
      "d2": undefined,
    },
  },
  {
    "a": 2,
    "b": false,
    "c": [
      "b",
    ],
    "d": {
      "d2": null,
    },
  },
  {
    "a": 2,
    "b": false,
    "c": [
      "b",
    ],
    "d": {
      "d2": undefined,
    },
  },
]
`;
